ui/app/src/Updates.elm

```diff
@@ -96,6 +99,64 @@ update msg model =
             in
                 ( { model | filter = { filter | text = t } }, Cmd.none )
 
+        KeyDownMsg code ->
+            let
+                mode =
+                    if List.member code modifierKeys then
+                        Append
+                    else
+                        Replace
+            in
+                ( { model | mode = mode }, Cmd.none )
+
+        KeyUpMsg code ->
+            let
+                mode =
+                    if List.member code modifierKeys then
+                        Replace
+                    else
+                        Append
+            in
+                ( { model | mode = mode }, Cmd.none )
+
+        AddLabel msg ( key, value ) ->
```

Comment:
I wasn't sure where to put it. It's definitely something only concerned with the `SilenceList` and `AlertList` views, but it is updating the top-level `filter`.

All the code for `labels` should be extracted into its own filter method, so it wouldn't be a problem to have the two different update functions implement this.

Thinking about this now, it would simplify the message being sent, and then the `update` function in question would know what follow-up message to use (fetch alerts/silences). I'll give it a try and see how it looks.